buildkite/scripts/unit-test.sh

```diff
@@ -14,6 +14,8 @@ source ~/.profile
 
 export MINA_LIBP2P_PASS="naughty blue worm"
 export NO_JS_BUILD=1 # skip some JS targets which have extra implicit dependencies
+export USE_LAGRANGE_CACHE=true
```

Comment:
Suggest rearranging the variable from `USE_LAGRANGE_CACHE` to `LANGRANGE_CACHE_DIR` and use it instead of `/tmp/lagrange_basis`

Also, a specific suggestion that might be useful: if a variable is provided and directory isn't writeable, do not try to attempt to write to cache, only read from it.

Reply:
@martyall it applies to nix context, not too important at this point though

For nix the directory will be provided that is extracted from tar, and writing to it will be prohibited

Reply:
I implemented the `LAGRANGE_CACHE_DIR` suggestion in another branch BTW

[georgeee/granular-nix-plus-proof-caching ef6f867913] Make cache dir configurable

Reply:
I just implemented it here as well